src/sparseml/core/session.py

```diff
@@ -269,8 +439,19 @@ def apply(
 
 
 class LifecycleCallbacks:
+    """
+    A class for invoking lifecycle events for the active session
+    """
+
```

Comment:
Clarify that these events get passed down to the modifiers, could put the lifecycle diagram of that here? But maybe thats better suited in the actual modifier class

Reply:
Putting it in the actual Modifier class!